Introduce writeMessage and onMessage APIs with data binding

Bhashinee opened the pull request:
## Purpose
Proposal: https://github.com/ballerina-platform/ballerina-standard-library/issues/2761

## Examples

## Checklist
- [x] Linked to an issue
- [x] Updated the specification
- [ ] Updated the changelog
- [x] Added tests

shafreenAnfar commented:
Just to clarify we only data binding and `anydata` support for `readMessage`, `writeMessage` and `onMessage`. Rest of the APIs can remain the same.

dilanSachi commented on docs/spec/spec.md:
```diff
@@ -3,53 +3,48 @@
 _Created_: 2021/12/09  
-_Updated_: 2022/02/18  
-_Edition_: Swan Lake  
+_Updated_: 2021/12/09  
```
```suggestion
_Updated_: 2022/05/05  
```

dilanSachi commented on native/src/main/java/io/ballerina/stdlib/websocket/WebSocketResourceDispatcher.java:
```diff
@@ -500,38 +503,35 @@ public static void dispatchOnText(WebSocketConnectionInfo connectionInfo, WebSoc
+                                    bValue = CloneWithType.convert(param,
+                                            StringUtils.fromString(stringAggregator.getAggregateString()));
+                                } else {
+                                    bValue = FromJsonStringWithType.fromJsonStringWithType(StringUtils.fromString(
```
we can send the else block to default case right?

Bhashinee replied:
Changed the code to fall through.

dilanSachi commented on native/src/main/java/io/ballerina/stdlib/websocket/WebSocketUtil.java:
```diff
@@ -218,6 +222,21 @@ static String getErrorMessage(Throwable err) {
+    }
+
+    public static boolean hasByteArrayType(Type targetType) {
+        List<Type> memberTypes = ((UnionType) targetType).getMemberTypes();
```
wont this accept types like `string[]` too?

Bhashinee replied:
Fixed with 12d7631

Bhashinee commented:
> Just to clarify we only need data binding and `anydata` support for `readMessage`, `writeMessage` and `onMessage`. Rest of the APIs can remain the same.

Addressed the suggestions. Removed the data binding support from `onTextMessage`, `readTextMessage` and `writeTextMessage` with [8f53776](https://github.com/ballerina-platform/module-ballerina-websocket/pull/587/commits/8f53776c5a0beda8c380e7df3350dcde7aeafc48)

dilanSachi reviewed (approved):
LGTM
